lib/xboxrt/libc_extensions/malloc.h

```diff
@@ -0,0 +1,10 @@
+#ifndef _XBOXRT_MALLOC_H_
+#define _XBOXRT_MALLOC_H_
+
+// Some code expects to find a declaration of malloc here, so we simply include the whole C header
+#include <stdlib.h>
+
+void *_alloca (size_t size);
+static void *alloca (size_t size) { return _alloca(size); }
```

Comment:
My own implementation in some apps used `void *__builtin_alloca (size_t size)` from https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html. I'm not sure about benefits or drawbacks, but would assume better error-checking at reduced compiler-compatibility.
Did you consider that?

Reply:
tbh I didn't expect that to exist. I've done some experimenting now and discovered that `_alloca` is a builtin, too, so it didn't even use the label I added in `chkstk.s`, but just emitted code that called `_chkstk`.

I modified the header to just forwards `alloca` to `_alloca` now.

Reply:
Did anyone test this patch? This should have been a `#define` - otherwise the `_alloca` runs in the local stack frame of this function and will be invalid by the time it returns.

Reply:
Good catch! Agree, we should make this a #define instead.